### ISSUE TITLE
IdsListView: Requestion Virtual scroll integration with ListView

### ISSUE THREAD

**Issue by vamsiprathipati** [infor-design/enterprise-wc#2322]
**Description**
I'm trying to integrate List view component with virtual scroll, I implemented virtual scrolling, but upon inspecting the elements in the developer tools, I noticed that all elements are still being loaded instead of just the visible ones.
 
**To Reproduce** 
 ```
 <ids-list-view>
<ids-virtual-scroll #virtualScroll1 height="100vh" id="virtual-scroll-1" item-height="20" buffer-size="3" item-count="1000"> 
<div class="ids-list-view"> 
<div class="ids-list-view-body" part="contents"> 
<ids-list-view-item>
// Required html code 
</ids-list-view-item>
</div>
</div>
</ids-virtual-scroll>
</ids-list-view>
```

**Steps to reproduce the behavior:**
1. Create a list view component
2. Add some list view items as shown above
3. Integrate virtual scroll with list-view
Expected behavior
Virtual scroll should be applied and only visible items are loaded
 
**Version**
ids-enterprise: ^1.0.0

**Comment by alixrosales** [comment]
@inforandy 
https://www.figma.com/design/RKVZ3yqiJqE6QfOlRbcUbK/List-%26-Header-Handoff---IDS-430-%2B-401?node-id=495-73433&t=LFfCscRf4yYNNVgK-4

**Comment by tmcconechy** [comment]
@vamsiprathipati @clayinfor 

I just noticed we have an example https://main.wc.design.infor.com/ids-list-view/virtual-scroll.html
However it doesnt seem to work too perfect.

b) i thought that the scroll behavior looks weird so tried to change the item height `item-height="95"` but that didnt work 
a) try scrolling down and you see the elements repeat

Source: https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/demos/virtual-scroll.ts and https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/demos/virtual-scroll.html

@vamsiprathipati do you have some special requirements outside this example about "loading" or not using the template or where unaware of this example. We can work on improving this example which might make this an easier task

**Comment by tmcconechy** [comment]
@vamsiprathipati please review the fix https://github.com/infor-design/enterprise-wc/pull/2486

**Comment by Varalaxmi26** [comment]
@tmcconechy , please add an angular example for list-view with the virtual scroll as we use this in the Angular project.

That would help us a lot.

Thanks in advance.

**Comment by tmcconechy** [comment]
@clayinfor can you add an NG example?

**Comment by Varalaxmi26** [comment]
Hi @tmcconechy, no NG example exists for using a virtual scroll in ListView with dynamic data. 

Our requirement is, 
The list view has to show two different types of template formats using a for loop based on the item type dynamically. 
The first template should show only the header.
The second template should show a combination of icons, text, and some buttons to perform some functional implementation. For this, we designed a component. 

We need a real-time use case example because our use case is different.
The current example https://main.wc.design.infor.com/ids-list-view/virtual-scroll.html is a static code example not helpful for us. 

Here is a sample code that we implemented. 
This code example should be shown in the virtual scroll.
When I added a virtual-scroll="true" to <ids-list-view> the list items, the empty screen was displayed. 

`<ids-list-view>
                        @for (item of listItems; track item.id) {
                            @if ((item | typeof) === "number") {
                                <ids-list-view-item class="group-header">
                                    <nc-item-group [groupNo]="item"> </nc-item-group>
                                </ids-list-view-item>
                            } @else {
                                <ids-list-view-item (click)="onRowClick(item)" [disabled]="item.disabled">
                                    //component code for row object which has some functional implementation. like, click and expand the row item.
                                </ids-list-view-item>
                            }
                        }
                    </ids-list-view>`

Screen shot:
![image](https://github.com/user-attachments/assets/9765fe7f-0cd8-4b29-a4a8-fe8541ad5b8f)
Please provide an example as early as possible.  That would be very helpful to us as it is a high-priority task in the Notification Center project.

I can arrange a call to explain the requirement If you need more clarification.

Thanks in Advance.

**Comment by tmcconechy** [comment]
@vamsiprathipati so you need to be able to use two templates at the same time? I dont think i understand that part of the requirement. So arrange a call im out next week so would have to be the following week.

https://main.wc.design.infor.com/ids-list-view/virtual-scroll.html isnt static you can adjust the template as needed 
https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/demos/virtual-scroll.html#L19-L21

You can also try to implement the template with markup like this example https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/demos/list-view-items-sortable.html#L18-L22 and put in an NG loop or something for it instead.

**Comment by Varalaxmi26** [comment]
yes. I have added like in https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/demos/list-view-items-sortable.html#L18-L22 using NG loop. 

Added  virtual-scroll="true" to ids-list-view. <ids-list-view virtual-scroll="true">. When ever I added this the list items were not displaying and an empty list was displayed like in bellow example.

![image](https://github.com/user-attachments/assets/5fa8b784-f895-45b9-9035-a248d9a7c4b6)

**Comment by Varalaxmi26** [comment]
The issue exists in angular-ids-wc examples also. I have added *virtual-scroll="true"* list view  <ids-list-view selectable="single" item-height="76"  **virtual-scroll="true"**> in angular-ids-wc/src/app/components/ids-list-view/demos/list-view-items/list-view-items.component.html. 
An empty screen was displayed after that.
![image](https://github.com/user-attachments/assets/f5ac2134-7d65-4bd1-aa43-c9a2586ebe1a)

**Comment by tmcconechy** [comment]
@clayinfor do you have time to add an angular example for the listview with virtual scroll?
@Varalaxmi26 one thing to note is for virtual scroll and when you set the item-height all list items should be that height. You might consider not using virtual scroll if you only have under 50,000 items and i think the speed should be fine.

**Comment by Varalaxmi26** [comment]
Hi @tmcconechy , for 1000 items we are getting out of memory exception in browser.
![image](https://github.com/user-attachments/assets/b89df908-0635-41fd-b5a4-b3a99459354f)

**Comment by tmcconechy** [comment]
@vamsiprathipati 
Hmm we might need a new issue for that? Can you make a reproduction so we can look? You mean without virtual scroll correct?

**Comment by Varalaxmi26** [comment]
Yes. This is without virtual scroll. In our application we are able to reproduce. Could you let us about know your availability.  I can arrange a call for discussion .

**Comment by Varalaxmi26** [comment]
Thanks for that. What about the issue mentioned in below comment?

https://github.com/infor-design/enterprise-wc/issues/2322#issuecomment-2228941110

Thanks

**Comment by tmcconechy** [comment]
@clayinfor would need an angular example for this.... do you think you can switch over and make that and confirm?

**Comment by Varalaxmi26** [comment]
Created https://github.com/infor-design/enterprise-wc/issues/2626 for memory issue.

**Comment by inforandy** [comment]
Reopening for clarification

**Comment by tmcconechy** [comment]
@Varalaxmi26 on this side of this issue we added an example https://github.com/infor-design/enterprise-wc-examples/tree/main/angular-ids-wc/src/app/components/ids-list-view/demos/virtual-scroll

It seems to work and we do not see this issue in your comment https://github.com/infor-design/enterprise-wc/issues/2322#issuecomment-2228941110

**Comment by Varalaxmi26** [comment]
Hi @tmcconechy , thanks for adding the example for virtual scroll in wc example.

We are adding the elements using ngFor in an HTML file like in the following example.
https://github.com/infor-design/enterprise-wc-examples/blob/main/angular-ids-wc/src/app/components/ids-list-view/demos/dynamic-disabled-items/dynamic-disabled-items.component.html

In this example if you add <ids-list-view selectable="single" item-height="76" **virtual-scroll="true"**></ids-list-view> you can see the issue which I was mentioned. 
![image](https://github.com/user-attachments/assets/7de34fb0-d202-48e4-a7b7-03e4a8fceb4f)

**Comment by tmcconechy** [comment]
@vamsiprathipati ok i see. Virtual scroll only works with a `data` set. You cannot use it when adding direct markup elements. This is a limitation with virtual scroll as it needs to know the data. (basically you would be adding all items to the DOM anyways so defeats the purpose). Try using a data set like the examples or leaving it off.

So that can never work. When doing that way i think you need to not use virtual scroll. Im working on an example with headers and markup based. With an append function to add more items. This might be the only way to do what you want (coming soon)
